lib/Lower/PFTBuilder.cpp

```diff
@@ -237,45 +239,514 @@ class PFTBuilder {
+              [&](const parser::ExitStmt *s) {
+                std::string name{getConstructName(s)};
+                pft::Evaluation *construct{name.empty()
+                                               ? doConstructStack.back()
+                                               : constructNameMap[name]};
+                assert(construct && "missing EXIT construct");
+                markBranchTarget(eval, *construct->constructExit);
+              },
+              [&](const parser::FlushStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::GotoStmt *s) { markBranchTarget(eval, s->v); },
+              [&](const parser::IfStmt *) { lastIfStmtEvaluation = &eval; },
+              [&](const parser::InquireStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::OpenStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::ReadStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::ReturnStmt *) { eval.isUnstructured = true; },
+              [&](const parser::RewindStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::StopStmt *) { eval.isUnstructured = true; },
+              [&](const parser::WaitStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::WriteStmt *s) { analyzeIoBranches(eval, s); },
+              [&](const parser::ComputedGotoStmt *s) {
+                for (auto &label : std::get<std::list<parser::Label>>(s->t)) {
+                  markBranchTarget(eval, label);
+                }
+              },
+              [&](const parser::ArithmeticIfStmt *s) {
+                markBranchTarget(eval, std::get<1>(s->t));
+                markBranchTarget(eval, std::get<2>(s->t));
+                markBranchTarget(eval, std::get<3>(s->t));
+              },
+              [&](const parser::AssignStmt *s) { // legacy label assignment
+                auto &label = std::get<parser::Label>(s->t);
+                auto sym = std::get<parser::Name>(s->t).symbol;
+                assert(sym && "missing AssignStmt symbol");
+                pft::Evaluation *t{labelEvaluationMap->find(label)->second};
+                if (!std::get_if<const parser::FormatStmt *const>(&t->u)) {
+                  markBranchTarget(eval, label);
+                }
+                auto iter = assignSymbolLabelMap->find(sym);
+                if (iter == assignSymbolLabelMap->end()) {
+                  pft::LabelSet labelSet{};
+                  labelSet.insert(label);
+                  assignSymbolLabelMap->try_emplace(sym, labelSet);
+                } else {
+                  iter->second.insert(label);
+                }
+              },
+              [&](const parser::AssignedGotoStmt *) {
+                // Specific control successors are not in general known.
+                // Compensate by directly marking the successor new block.
+                markSuccessorNewBlock(eval);
```

Comment:
This call is needed because assigned gotos don't in general have known goto targets.  I've changed the comment as below.  Another option might be to arrange to pick some candidate target for the goto's control successor, but doing that seems rather odd and unnecessary.

            // Although this statement is a branch, it doesn't have any
            // explicit control successors.  So the code at the end of the
            // loop won't mark the exit successor.  Do that here.

Reply:
Thanks, I get it now